src/Bicep.Core.IntegrationTests/Scenarios/ParamKeyVaultSecretReferenceTests.cs

```diff
@@ -378,6 +379,121 @@ public void NoDiagnosticsForModuleWithABadPath()
 
             result.Should().NotGenerateATemplate();
             result.Should().NotHaveDiagnosticsWithCodes(new[] { "BCP180" }, "Function placement should not be evaluated on a module that couldn't be read.");
+            
+        }
+
+        /// <summary>
+        /// https://github.com/Azure/bicep/issues/4270
+        /// </summary>
+        [TestMethod]
+        public void SecretsWithConditional()
+        {
+            var result = CompilationHelper.Compile(
+                ("main.bicep", @"
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+module secret 'secret.bicep' = {
+  name: 'secret'
+  params: {
+    mySecret: true == true ? kv.getSecret('mySecret','secretversionguid') : ''
+  }
+}
+"), ("secret.bicep", @"
+@secure()
+param mySecret string = 'defaultSecret'
+
+output exposed string = mySecret
+"));
+
+            result.Should().GenerateATemplate().And.NotHaveAnyDiagnostics();
+            var parameterToken = result.Template!.SelectToken("$.resources[?(@.name == 'secret')].properties.parameters.mySecret")!;
+            using (new AssertionScope())
+            {
+                parameterToken.Value<string>().Should().Be("[if(equals(true(), true()), createObject('reference', createObject('keyVault', createObject('id', resourceId('Microsoft.KeyVault/vaults', 'testkeyvault')), 'secretName', 'mySecret', 'secretVersion', 'secretversionguid')), createObject('value', ''))]");
+            }
+        }
+        
+        [TestMethod]
+        public void SecretsWithConditional_InNestedCondition()
+        {
+            var result = CompilationHelper.Compile(
+                ("main.bicep", @"
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: 'testkeyvault'
+}
+
+module secret 'secret.bicep' = {
+  name: 'secret'
+  params: {
+    mySecret: true == true ? false == false ? kv.getSecret('mySecret','secretversionguid') : 'false' : 'true'
```

Comment:
Could you add a test to verify this also works with parentheses?
`mySecret: true == true ? (false == false ? kv.getSecret('mySecret','secretversionguid') : 'false') : 'true'`